### ISSUE TITLE
Topology matcher?

### ISSUE THREAD

**Issue by mjy** [OpenTreeOfLife/hackathon#22]
Sorry, haven't done much homework on this, but is there a OT API mechanism to supply a topology only, and get back matching clades that exactly match that topology?

**Comment by josephwb** [comment]
Nope, not even close.

**Comment by chinchliff** [comment]
You just mean provide a tree structure and get back clades that have identical tree structure? Assuming you don't mean branch lengths, since we don't have many of those. I'm curious, what is the use case for this?

**Comment by bomeara** [comment]
I could see a use case using an incompletely resolved phylogeny: show me
all trees with a gnetales+pine clade, or where humans are sister to
orangutans (yes, this is crazy, but if any of you have been on taxacom for
awhile, you'll have seen this....).

**Comment by chinchliff** [comment]
Right, that makes sense. Given a tree structure with tips assigned to taxa, find all known trees that are consistent with that tree structure for those taxa.

We definitely don't have this. I think treemachine (the tree graph builder tool) would be the place for it to go, but it would be limited (of course) to the trees that had been imported from the treestore. Currently we use treemachine primarily for building the draft tree, and we only import the trees that we need for that (a relatively small subset, since many are redundant and/or rather full of errors). We could have another instance of treemachine for these kinds of queries, or we could reimplement the method in a different program. I do think the treemachine graph structure would lend itself well to this analysis.

I guess I had interpreted the original question here as "given a tree structure only, _without any mapped taxa_, return a set of clades in the tree that have this structure". I'm still not sure what the use case is for that, or maybe I misinterpreted that question?

**Comment by bomeara** [comment]
There was at some point some query language (phylows? some *ql thing?) that
allowed for queries of this sort. I could see limited use cases: "I need a
pectinate but empirical tree to test my method" sort of thing, but that's
rare enough that I wouldn't bother addressing it.

**Comment by mjy** [comment]
Wouldn't there be interest in seeing if there is a bias in topology
patterns within the tree of life?  To test this you'd need this
functionality.

**Comment by josephwb** [comment]
Unfortunately, much of the tree topology comes from taxonomy. Is the taxonomy structure biased? Probably!

**Comment by bomeara** [comment]
There have been studies of that sort, but they take a topology and
calculate a metric from it. Doing it by counting the matching tree shapes
to (,(,(,(,())))) vs (((,),),()) and so forth would be an odd way to do it.
There are MANY possible tree shapes, even without tip labels, between a
pectinate and a balanced tree, and so searching for each as a way to get at
tree balance is inefficient.

**Comment by mjy** [comment]
Thanks for the explanation, that makes sense.  Re topology from taxonomy-
perhaps there is an interesting question right there, do taxonomists come
up with topologies of a certain "type" (metric) and then do these match
what is found via phylogenetic methods.

Slight change of topic-  is it safe to assume there is a method to return
branch annotations indicating the branch comes from from taxonomy rather
than phylogenetic inference?

**Comment by josephwb** [comment]
@mjy Yes, we have the information about whether branches are supported by taxonomy and/or tree(s).
